runners/direct-java/src/main/java/org/apache/beam/runners/direct/AggregatorContainer.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.direct;
+
+import org.apache.beam.sdk.transforms.Aggregator;
+import org.apache.beam.sdk.transforms.Aggregator.AggregatorFactory;
+import org.apache.beam.sdk.transforms.Combine.CombineFn;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * AccumT container for the current values associated with {@link Aggregator Aggregators}.
+ */
+public class AggregatorContainer {
+
+  private static class AggregatorInfo<InputT, AccumT, OutputT>
+      implements Aggregator<InputT, OutputT> {
+    private final String name;
+    private final CombineFn<InputT, AccumT, OutputT> combiner;
+    private AccumT accumulator = null;
+    private boolean committed = false;
+
+    private AggregatorInfo(String name, CombineFn<InputT, AccumT, OutputT> combiner) {
+      this.name = name;
+      this.combiner = combiner;
+    }
+
+    @Override
+    public void addValue(InputT input) {
+      Preconditions.checkState(!committed, "Cannot addValue after committing");
+      if (accumulator == null) {
+        accumulator = combiner.createAccumulator();
+      }
+      accumulator = combiner.addInput(accumulator, input);
+    }
+
+    public OutputT getOutput() {
+      return accumulator == null ? null : combiner.extractOutput(accumulator);
+    }
+
+    private void merge(AggregatorInfo<?, ?, ?> other) {
+      // TODO: Make sure the CombineFn and accumulators are compatible.
```

Comment:
Aggregators should only be merged if they are "the same" - Aggregators are named by the DoFnRunner per-step, and the name is unique per step, so name should suffice here.